### PR TITLE
FLW-786 Unbond flow: Confirmation screen

### PR DESCRIPTION
**Pull request by ERussel**
- add unbond confirmation screen

### CODE REVIEW

File: fearless/Modules/Staking/StakingUnbondConfirm/StakingUnbondConfirmPresenter.swift [hunk 1]
```diff
@@ -0,0 +1,289 @@
...
+
+    func refreshFeeIfNeeded() {
+        guard fee == nil, controller != nil, payee != nil, bonded != nil, minimalBalance != nil else {
+            return
+        }
+
+        interactor.estimateFee(for: inputAmount, resettingRewardDestination: shouldResetRewardDestination)
+    }
+
+    init(
```
Comment (stas-litvinov) [inline fearless/Modules/Staking/StakingUnbondConfirm/StakingUnbondConfirmPresenter.swift:84]: It's rather unusual to see an initialiser after public func

File: fearless/Modules/Staking/StakingUnbondConfirm/StakingUnbondConfirmInteractor.swift [hunk 2]
```diff
@@ -0,0 +1,241 @@
...
+    let singleValueProviderFactory: SingleValueProviderFactoryProtocol
+    let substrateProviderFactory: SubstrateDataProviderFactoryProtocol
+    let settings: SettingsManagerProtocol
+    let runtimeService: RuntimeCodingServiceProtocol
+    let operationManager: OperationManagerProtocol
+    let extrinsicServiceFactory: ExtrinsicServiceFactoryProtocol
+    let feeProxy: ExtrinsicFeeProxyProtocol
+    let accountRepository: AnyDataProviderRepository<AccountItem>
+    let chain: Chain
+    let assetId: WalletAssetId
```
Comment (stas-litvinov) [inline fearless/Modules/Staking/StakingUnbondConfirm/StakingUnbondConfirmInteractor.swift:19]: Looks like everything here should be private

File: fearless/Common/View/UIFactory.swift [hunk 3]
```diff
@@ -43,9 +43,21 @@ protocol UIFactoryProtocol {
...
+
+    func createNetworkFeeConfirmView() -> NetworkFeeConfirmView
+
+    func createTitleValueView() -> TitleValueView
+
+    func createHintView() -> HintView
 }
 
 final class UIFactory: UIFactoryProtocol {
+    static let `default` = UIFactory()
```
Comment (stas-litvinov) [inline fearless/Common/View/UIFactory.swift:59]: Instead of using reserved word I would suggest to rename it to defaultFactory

**Review by stas-litvinov — APPROVED** [review]

File: fearless/Modules/Staking/StakingUnbondConfirm/StakingUnbondConfirmInteractor.swift (hunk 2 above)
Comment (ERussel) [reply fearless/Modules/Staking/StakingUnbondConfirm/StakingUnbondConfirmInteractor.swift:19]: We just need a rule for this. In my opinion it is convenient to have public properties which just were passed as parameters in init.

File: fearless/Common/View/UIFactory.swift (hunk 3 above)
Comment (ERussel) [reply fearless/Common/View/UIFactory.swift:59]: But think about usage: ```UIFactory.default``` vs ```UIFactory.defaultFactory```

**Pull request merged**